storage/statedb/trie_test.go

```diff
@@ -371,6 +371,7 @@ func TestPruningByUpdate(t *testing.T) {
 	// By inserting "dogglesworth", both extension and branch nodes are affected, hence pruning the both.
 
 	// Update and commit to store the nodes
+	trie.originalRoot = nodehash1 // set trie root hash
```

Comment:
Instead of changing an internal variable here, let's reopen a Trie at line 384.
You can change the line 422 similarly.